client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala

```diff
@@ -65,42 +62,26 @@ object FollowRedirect {
           else
             req
 
-        def nextRequest(method: Method, nextUri: Uri, bodyOpt: Option[Stream[F, Byte]])
-          : Request[F] =
-          bodyOpt match {
-            case Some(body) =>
-              stripSensitiveHeaders(nextUri)
-                .withMethod(method)
-                .withUri(nextUri)
-                .withBodyStream(body)
-            case None =>
-              stripSensitiveHeaders(nextUri)
-                .withMethod(method)
-                .withUri(nextUri)
-                .withEmptyBody
-          }
-
         def doRedirect(method: Method): Resource[F, Response[F]] =
           if (redirects < maxRedirects) {
             // If we get a redirect response without a location, then there is
             // nothing to redirect.
             redirectUri.fold(dontRedirect) { nextUri =>
               // We can only redirect safely if there is no body or if we've
               // verified that the body is pure.
```

Comment:
This comment is obsolete.  We've given up determining that the body is pure.  That's something we did on previous versions of the streaming library where we could introspect the algebra, which is where a lot of of this old complication came from in the first place.